### PR TITLE
Feature/toolmanagement

### PR DESCRIPTION
**Pull request by vikramjadon921995**
#### Why
Tool management APIs are required for managing employees, tools, tool types and their tool transactions from the mobile application being used at sites.

#### Changes
+ Added CRUD and test cases for Employee.
+ Added CRUD and test cases for Tooltype.
+ Added CRUD and test cases for Toolbox.
+ Added CRUD and test cases for Tool.
+ Added API endpoint, logic and test cases for Tool transaction.

### CODE REVIEW

File: apps/acqdat_api/test/acqdat_api_web/controllers/tool_management_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,298 @@
+defmodule AcqdatApiWeb.ToolManagementControllerTest do
```
Comment (arjun289) [inline apps/acqdat_api/test/acqdat_api_web/controllers/tool_management_controller_test.exs]: After moving the routes to be protected under `authenticated`, please add some tests here for unauthenticated access on those routes.

File: apps/acqdat_api/lib/acqdat_api/context/sensor.ex [hunk 2]
```diff
@@ -1,8 +1,8 @@
 defmodule AcqdatApi.Context.Sensor do
   alias AcqdatCore.Model.Sensor, as: SensorModel
 
-  def sensor_by_criteria(%{"device_id" => device_id} = criteria) do
+  def sensor_by_criteria(%{"device_id" => device_id} = _criteria) do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api/context/sensor.ex]: Done

File: apps/acqdat_api/lib/acqdat_api/context/tool_management.ex [hunk 3]
```diff
@@ -0,0 +1,163 @@
+defmodule AcqdatApi.Context.ToolManagement do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api/context/tool_management.ex]: Done

File: apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_box_controller.ex [hunk 4]
```diff
@@ -0,0 +1,103 @@
+defmodule AcqdatApiWeb.ToolManagement.ToolBoxController do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_box_controller.ex]: Done

File: apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_type_controller.ex [hunk 5]
```diff
@@ -0,0 +1,103 @@
+defmodule AcqdatApiWeb.ToolManagement.ToolTypeController do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_type_controller.ex]: Done

File: apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_controller.ex [hunk 6]
```diff
@@ -0,0 +1,138 @@
+defmodule AcqdatApiWeb.ToolManagement.ToolController do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/tool-management/tool_controller.ex]: Done

File: apps/acqdat_api/lib/acqdat_api_web/router.ex [hunk 7]
```diff
@@ -27,13 +27,41 @@ defmodule AcqdatApiWeb.Router do
...
+    resources "/device", DeviceController, only: [:create, :update, :delete, :index, :show]
     get "/sensor-criteria/:device_id", SensorController, :sensor_by_criteria
+    resources "/sensor", SensorController, only: [:create, :update, :delete, :index, :show]
     resources "/policies", NotificationPolicyController, only: [:index]
 
     resources "/sensor_notification", SensorNotificationController,
       only: [:create, :update, :delete, :index, :show]
   end
+
+  scope "/api", AcqdatApiWeb do
```
[the remark this answers is not in the document]
Comment (Ayoush) [reply apps/acqdat_api/lib/acqdat_api_web/router.ex]: Done

File: apps/acqdat_api/test/acqdat_api_web/controllers/tool_management_controller_test.exs (hunk 1 above)
Comment (Ayoush) [reply apps/acqdat_api/test/acqdat_api_web/controllers/tool_management_controller_test.exs]: Done

**Pull request merged**